### PR TITLE
Fix RTL text issue in the inspector in firefox in GiJoe

### LINKED ISSUE #1671: Entering text into inspector text fields is going right to left
In the picture see the entry for `debug` in the inspector where typing "yes" resulted in "sey".

2016-05-17-091530_1559x714_scrot

### PR DESCRIPTION
**Pull request by hatched**
Fixes #1671

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA
Deploy with GiJoe and add the wordpress charm to the canvas and check that the inspector config values type the correct direction in FireFox

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5110/
Test PASSed.

**Comment by frankban** [comment]
:+1:

**Comment by huwshimi** [comment]
I hate to say it, but I'm still getting this with the wordpress charm in Firefox (using the sandbox).

**Comment by hatched** [comment]
This DID fix it, but I now also see that it's no longer working. I'll investigate further.

**Comment by hatched** [comment]
@huwshimi Can you take another look please. The issue turned out to be caused by FireFox not maintaining cursor position when the editable field was re-rendered by React. So by preventing the re-render the cursor now maintains position across all the browsers.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5117/
Test PASSed.

File: jujugui/static/gui/src/app/components/string-config/string-config.js [hunk 1]
```diff
@@ -30,6 +30,11 @@ YUI.add('string-config', function() {
       this.setState({ value: nextProps.config });
     },
 
+    shouldComponentUpdate: function(nextProps, nextState) {
+      return nextState.value !==
+        this.getDOMNode().querySelector('.string-config--value').innerText;
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/string-config/string-config.js]: I think `getDOMNode` is deprecated, I think you want `ReactDOM.findDOMNode(this)`, or whatever the correct new syntax is.

File: jujugui/static/gui/src/app/components/string-config/test-string-config.js [hunk 2]
```diff
@@ -60,6 +60,16 @@ describe('StringConfig', function() {
     assert.deepEqual(output, expected);
   });
 
+  function stubGetDomNode(instance) {
+    instance.getDOMNode = _ => {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/string-config/test-string-config.js]: This will have to be updated once you use findDOMNode.

**Comment by huwshimi** [comment]
Much better @hatched. QA OK. :+1: with the new method of getting the DOM node.

File: jujugui/static/gui/src/app/components/string-config/string-config.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/string-config/string-config.js]: Good catch! Thanks

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5123/
Test PASSed.

**Comment by hatched** [comment]
Thanks all! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**